### PR TITLE
Add common terms to the shared glossary

### PR DESCRIPTION
**Pull request by jakobw**
Bug: [T272841](https://phabricator.wikimedia.org/T272841)

### CODE REVIEW

File: systems/overview/12-Glossary.md [hunk 1]
```diff
@@ -26,14 +26,34 @@ A component is a software unit with a well-defined interface and explicitly spec
...
 Wikibase Extensions are MediaWiki extensions that add further functionality to Wikibase. They depend on the Wikibase extension being installed and directly interact with it.
 Examples: WikibaseLexeme, WikibaseMediaInfo, WikibaseImport
 
+## Entity
+
+Entities are the top level concepts of Wikibase's data model. Items and Properties are the core Entity types of Wikibase. Other types can be added through [Wikibase extensions](#wikibase-extension), such as [Lexemes](https://www.mediawiki.org/wiki/Extension:WikibaseLexeme/Data_Model#Lexeme).
+
+## [Item](https://www.wikidata.org/wiki/Help:Items)
+
+In Wikidata, items are used to represent all the things in human knowledge, including topics, concepts, and objects. For example, the "1988 Summer Olympics", "love", "Elvis Presley", and "gorilla" are all items in Wikidata.
```
Comment (addshore) [inline systems/overview/12-Glossary.md:48]: items -> Items?

Comment (jakobw) [reply systems/overview/12-Glossary.md]: Indeed! Fixed. Now I need to figure out where I copy-pasted that from and fix it there too :D

File: systems/overview/12-Glossary.md [hunk 2]
```diff
@@ -26,14 +26,34 @@ A component is a software unit with a well-defined interface and explicitly spec
 Extensions are software components that administrators can add (or remove) to their MediaWiki application in order to customize how it looks and works.
 Examples: WikibaseClient, UniversalLanguageSelector, OAuth
 
+## [WikibaseClient](https://www.mediawiki.org/wiki/Extension:Wikibase_Client)
+
+WikibaseClient is a [MediaWiki Extension](#mediawiki-extension) that enables the use and display of Entity data from a [Wikibase Repository](#wikibase-repository) via Lua modules or parser functions. Clients can also use centralized language links.
+
+## Wikibase Repository
```
Comment (tzhelyazkova) [inline systems/overview/12-Glossary.md:33]: Maybe we can also link to the extension page in the title.

Comment (jakobw) [reply systems/overview/12-Glossary.md:33]: I deliberately linked to the extension page only from the description but not from the title to make it clearer that we're talking about [the Repo application, not the extension](https://github.com/wmde/wikidata-wikibase-architecture/blob/main/docs/decisions/client-repo-mediawiki.md) here. Not sure if that needs to be made clearer here or if it's fine to link to it here anyway.

Comment (tzhelyazkova) [reply systems/overview/12-Glossary.md:33]: Oh ok, then maybe we can add a sentence to the description saying "It can optionally contain WikibaseLexeme etc." ?

File: systems/overview/12-Glossary.md [hunk 3]
```diff
@@ -26,14 +26,34 @@ A component is a software unit with a well-defined interface and explicitly spec
...
+
+Entities are the top level concepts of Wikibase's data model. Items and Properties are the core Entity types of Wikibase. Other types can be added through [Wikibase extensions](#wikibase-extension), such as [Lexemes](https://www.mediawiki.org/wiki/Extension:WikibaseLexeme/Data_Model#Lexeme).
+
+## [Item](https://www.wikidata.org/wiki/Help:Items)
+
+In Wikidata, Items are used to represent all the things in human knowledge, including topics, concepts, and objects. For example, the "1988 Summer Olympics", "love", "Elvis Presley", and "gorilla" are all Items in Wikidata.
+
+## [Property](https://www.wikidata.org/wiki/Help:Properties)
+
+Properties are Entities that describe a relationship between Items (or other Entities) and Values of the property. Typical properties are population (using numbers as values), binomial name (using strings as values), but also has father and author of (both using Items as values).
```
Comment (Silvan-WMDE) [inline systems/overview/12-Glossary.md:52]: 1. I spot two lowercase properties here. 
2. Which of the terms that have a Glossary entry should be linked from other Glossary entries? I see Entities, Items and Values here. (wait, Value doesn't even have an entry, although it's capitalised...)
3. I think the property names should be marked somehow, either in quotes or in italic, otherwise the sentence is quite hard to parse.

File: systems/overview/12-Glossary.md (hunk 2 above)
Comment (jakobw) [reply systems/overview/12-Glossary.md:33]: Yup that makes sense. Added.

**Review by tzhelyazkova — APPROVED** [review]

File: systems/overview/12-Glossary.md (hunk 3 above)
Comment (addshore) [reply systems/overview/12-Glossary.md]: I rebased this as I was about to merge it, then imediatly saw this comment so also tackeled some of this in https://github.com/wmde/wikidata-wikibase-architecture/pull/57/commits/343499dd002d53d6249b3e9f5b8e9a626e176f9a
We should come up with a guide on when we should and shouldn't link to the glossary.

**Review by addshore — APPROVED** [review]

File: systems/overview/12-Glossary.md (hunk 3 above)
Comment (jakobw) [reply systems/overview/12-Glossary.md]: Thanks! FWIW this was copied from here https://www.mediawiki.org/wiki/Wikibase/DataModel#Properties
